app/jobs/irs_attempts_events_batch_job.rb

```diff
@@ -5,6 +5,16 @@ def perform(timestamp = Time.zone.now - 1.hour)
     enabled = IdentityConfig.store.irs_attempt_api_enabled && s3_helper.attempts_s3_write_enabled
     return nil unless enabled
 
+    previous_hour = timestamp - 1.hour
+    # Check if previous hour was properly loaded
+    if (
+      !IrsAttemptApiLogFile.find_by(requested_time: timestamp_key(key: previous_hour)) &&
+      reasonable_timespan(previous_hour)
+    )
```

Comment:
is there a way we could put this on one line? or maybe make a helper?

```suggestion
    if missing_file?(previous_hour)
```

Also: should we have a config that checks the last X hours? For example with a 2-hour downtime, we'd only pick up one missed job?

Reply:
Yes, I can put that in a helper method. 👍🏽

Since this is recursive, in the case of a multi-hour gap, it will keep going back an hour until it fills the whole gap.  That's why we put in the 'reasonable_time' stop-gap (@n1zzy's idea!).